### PR TITLE
Implement MeshCNN

### PR DESCRIPTION
**Pull request by krrish94**
* Implement MeshCNN in Kaolin
* Also port a fix for adjacency-info from @Caenorst 

Signed-off-by: Krishna Murthy <krrish94@gmail.com>

### CODE REVIEW

File: kaolin/models/meshcnn.py [hunk 1]
```diff
@@ -0,0 +1,994 @@
...
+import torch.nn.functional as F
+
+__all__ = [
+    "MeshCNNClassifier",
+    "compute_face_normals_and_areas",
+    "extract_meshcnn_features",
+]
+
+
+def compute_face_normals_for_mesh(mesh):
```
Comment (Caenorst) [inline kaolin/models/meshcnn.py:54]: This function have a very similar name than https://github.com/NVIDIAGameWorks/kaolin/blob/83002fedc67e1c9112a8f834ffb4f8a890e6042a/kaolin/rep/TriangleMesh.py#L154 but a different computation (can you explain ?)

File: kaolin/models/meshcnn.py [hunk 2]
```diff
@@ -0,0 +1,994 @@
...
+
+def compute_face_normals_and_areas(mesh):
+    r"""Compute face normals and areas for an input kaolin.rep.TriangleMesh object.
+    """
+    face_normals = torch.cross(
+        mesh.vertices[mesh.faces[:, 1]] - mesh.vertices[mesh.faces[:, 0]],
+        mesh.vertices[mesh.faces[:, 2]] - mesh.vertices[mesh.faces[:, 1]],
+    )
+    face_normal_lengths = face_normals.norm(p=2, dim=-1)
+    face_normals = face_normals / face_normal_lengths[..., None]
```
Comment (Caenorst) [inline kaolin/models/meshcnn.py:92]: `unqueeze(-1)` at line 61 and `[..., None]` here ?

File: tests/examples/run/test_classification.sh [hunk 3]
```diff
@@ -8,3 +8,8 @@ python ${KAOLIN_HOME}/examples/Classification/pointcloud_classification.py \
     --modelnet-root /data/ModelNet10/ \
     --categories desk dresser \
     --epochs 1 --transforms-device cpu
+
+python ${KAOLIN_HOME}/examples/Classification/mesh_classification.py \
```
Comment (Caenorst) [inline tests/examples/run/test_classification.sh:12]: In tests the CI catch the returned value of the script, if you run those two commands like that if the first command fails the return value is gonna be overriden by this one.

I recommend using command1 && command2

**Review by Caenorst — CHANGES_REQUESTED** [review]
Pre-review.

On kaolin/models/meshcnn.py:
We need to have documentation on all the functions (especially for the arguments).
We need to have unit tests for all those functions.
I know is currently not enforced but please avoid using raw string in documentation if you don't need it.

File: kaolin/models/meshcnn.py (hunk 1 above)
Comment (krrish94) [reply kaolin/models/meshcnn.py:54]: As the comment in `compute_face_normals` in `rep/TriangleMesh.py` says, we have a "more robust" normal computation scheme, where we average normals for each vertex to compute the face normal.

Here, I've chosen this computation (as opposed to Kaolin's impl) to exactly align with the feature computation scheme employed in MeshCNN.

**Comment by krrish94** [comment]
Also, for the docstrings, I currently added docstrings only for functions that are exposed externally. All other functions are used only within this file.

Still, I think it's reasonable to add docstrings and unittests for all modules regardless. Will do

**Review by Caenorst — APPROVED** [review]
LGTM !

**Pull request merged**